cmake/Dependencies.cmake

```diff
@@ -1156,6 +1156,18 @@ if(APPLE)
   target_link_options(pybind::pybind11 INTERFACE -undefined dynamic_lookup)
 endif()
 
+# ---[ OpenTelemetry API headers
+if(NOT OpenTelemetryApi_FOUND)
```

Comment:
You don't seem to ever find the package, so this var will never be set, so you're always going to use the third party.

I actually think it's OK to always use third party. But the if branch here is very misleading.